styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -141,24 +142,38 @@ void tick() {
       if (resourceRefs.isEmpty()) {
         sendDequeue(instance);
       } else {
-        final boolean canRun = resourceRefs.stream()
-            .allMatch(resourceId -> {
+        final Set<String> unknownResources = Sets.newHashSet();
```

Comment:
Maybe it's time for this whole branch to be broken out to it's own method. There's too much branching here now.